packages/uploadthing/src/client.ts

```diff
@@ -84,32 +54,77 @@ type UploadFilesOptions<TRouter extends FileRouter> = {
 }[keyof TRouter];
 
 export type UploadFileResponse = {
-  /**
-   * @deprecated
-   * use `name` instead
-   */
-  fileName: string;
   name: string;
-  /**
-   * @deprecated
-   * use `size` instead
-   */
-  fileSize: number;
   size: number;
-  /**
-   * @deprecated
-   * use `key` instead
-   */
-  fileKey: string;
   key: string;
-  /**
-   * @deprecated
-   * use `url` instead
-   */
-  fileUrl: string;
   url: string;
 };
 
+async function alertUTOfError(opts: {
+  url?: string;
+  endpoint: string;
+  key: string;
+  uploadId: string;
+  fileName: string;
+  reason?: string;
+}) {
+  // tell uploadthing infra server that upload failed
+  await fetch(
+    createAPIRequestUrl({
+      url: opts.url,
+      slug: opts.endpoint,
+      actionType: "failure",
+    }),
+    {
+      method: "POST",
+      body: JSON.stringify({
+        fileKey: opts.key,
+        uploadId: opts.uploadId,
+      }),
+    },
+  );
+
+  // Attempt to parse response as XML
+  const parsed = maybeParseResponseXML(opts.reason ?? "");
+
+  // Throw an error for the client
+  if (parsed?.message) {
+    throw new UploadThingError({
+      code: parsed.code,
+      message: parsed.message,
+    });
+  } else {
+    throw new UploadThingError({
+      code: "UPLOAD_FAILED",
+      message: `Failed to upload file ${opts.fileName} to S3`,
+      cause: opts.reason,
+    });
+  }
+}
+
+async function alertUTOfUploadComplete(opts: {
```

Comment:
Really don't dig the `alert` prefix for these. Also crazy thought, maybe group them all into one object just for naming's sake?

Reply:
Went a bit further https://github.com/pingdotgg/uploadthing/compare/b817f2a0e53e0f33674c17efefa671b84d041939..b4ee21ce96ecef62b025152f0afafd252e236a5f